src/Microsoft.Azure.WebJobs.Host/Extensions/ExceptionExtensions.cs

```diff
@@ -1,15 +1,76 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
 using System.Reflection;
 using System.Runtime.InteropServices;
+using System.Text;
 using System.Threading;
 using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Diagnostics;
+using Microsoft.WindowsAzure.Storage;
 
 namespace System
 {
     internal static class ExceptionExtensions
     {
+        public static string ToDetails(this Exception exception)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException("exception");
+            }
+
+            // Additional details are only available on StorageException.
+            IDictionary<string, string> additionalDetails = GetAdditionalDetails(exception);
+
+            if (additionalDetails == null || additionalDetails.Count == 0)
+            {
+                return ExceptionFormatter.GetFormattedException(exception);
```

Comment:
This now uses the `GetFormattedException` method as opposed to just calling `ToString`